### PR TITLE
Attestation: Track and measure the TVM configuration

### PR DESCRIPTION
**Pull request by sameo**
This PR tracks the TVM configuration bits (EPC and initial argument) and extends PCR3 with a measurement of that data when the TVM finalizes.

In order to include this measurement into the right PCR, this PR also includes a PCR mapping update and fix.

### CODE REVIEW

File: attestation/src/measurement.rs [hunk 1]
```diff
@@ -78,15 +58,90 @@ enum CsrCdi {}
...
+    // to `extend` fail.
+    static_measurement: bool,
+
+    // For a given workload, a stable measurement register must not change
+    // between boots. If it does change, this means the workload TCB actually
+    // changed. For example, a platform config measurement is not stable, while
+    // an OS one is.
+    // A sealing CDI must only be derived from stable measurements, as we don't
+    // want sealed data to depend on layers that could change between platforms.
+    // An attestation CDI must include non stable measurements.
```
Comment (atulkharerivos) [inline attestation/src/measurement.rs:80]: Nit: non-stable

File: attestation/src/measurement.rs [hunk 2]
```diff
@@ -78,15 +58,90 @@ enum CsrCdi {}
...
+
+macro_rules! msmt_dynamic_reg {
+    ($fw_index:expr, $pcr_index:expr, $stable:expr) => {{
+        msmt_reg!($fw_index, $pcr_index, false, $stable)
+    }};
+}
+
+const TVM_MSMT_REGISTERS: [MeasurementRegisterBuilder; MSMT_REGISTERS] = [
+    // TCG PCR 0 - Platform trusted firmware code.
+    // This is comming from the hardware RoT.
```
Comment (atulkharerivos) [inline attestation/src/measurement.rs:124]: s/.../This comes/...

File: attestation/src/measurement.rs [hunk 3]
```diff
@@ -114,6 +169,9 @@ struct MeasurementRegister<D: Digest> {
     // between boots. If it does change, this means the workload TCB actually
     // changed. For example, a platform config measurement is not stable, while
     // an OS one is.
+    // A sealing CDI must only be derived from stable measurements, as we don't
+    // want sealed data to depend on layers that could change between platforms.
+    // An attestation CDI must include non stable measurements.
```
Comment (atulkharerivos) [inline attestation/src/measurement.rs:174]: Nit: non-stable

File: attestation/src/measurement.rs [hunk 4]
```diff
@@ -78,15 +58,90 @@ enum CsrCdi {}
...
-    19, // TCG PCR 19 - DRTM
-    20, // TCG PCR 20 - DRTM
+struct MeasurementRegisterBuilder {
+    // The DiceTcbInfo FWIDs list index.
+    fwid_index: u8,
+
+    // TCG PCR index.
+    pcr_index: u8,
+
+    // A static register can not be extended after the platform is booted, i.e.
```
Comment (atulkharerivos) [inline attestation/src/measurement.rs:68]: Nit: cannot

File: attestation/src/measurement.rs [hunk 5]
```diff
@@ -78,15 +58,90 @@ enum CsrCdi {}
...
+    // after the static TCB is finalized.
+    // A call to `finalize` will lock static registers, making subsequent calls
+    // to `extend` fail.
+    static_measurement: bool,
+
+    // For a given workload, a stable measurement register must not change
+    // between boots. If it does change, this means the workload TCB actually
+    // changed. For example, a platform config measurement is not stable, while
+    // an OS one is.
```
Comment (atulkharerivos) [inline attestation/src/measurement.rs:77]: Nit: Consider rewording without the trailing "is". Perhaps: Examples of stable measurements include <blah>, but <blah> is non-stable.

**Review by atulkharerivos — APPROVED** [review]
LGTM; just some nits about wording.

**Review by dgreid — APPROVED** [review]

**Review by abrestic-rivos — APPROVED** [review]

**Comment by sameo** [comment]
@atulkharerivos Thanks for the feedback, all comments are addressed now.

**Pull request merged**